### PR TITLE
Ux 412 Developer sees the log broken down by step

### PR DESCRIPTION
**Pull request by scherler**
Related to issue # UX-412. 

Summary of this pull request: 
.
.* For any node in the railway the developer can click on it and see a list of the steps that had executed for that node (x)
* Each step has a status (passed, failed or running)  (/)
* Clicking on one of the steps expands the log in place. (/)
* On successful, aborted, unstable and unknown runs the developer sees all the steps in the log collapsed (/)
* On failed run the developer sees all the steps in the log collapsed but the failing step is expanded (/)
* On executing run the developer sees all the steps in the log collapsed but the currently executing step is expanded (/)
* Not in the mockup: 2px radius on the top and bottom un-expanded steps
* Don't fetch the whole log into the browser memory ever (break the "download log" feature if needed) (/)
* Show full log in freestyle cases (no step by step break down) but without eagerly loading the whole multi MB log (should only load last 4K (ping Vivek), as jenkins console does now, with a link to "show full log" that then loads the rest. (!) not test with other then pipeline and multibranch
.

@reviewbybees 

screenshot from 2016-05-27 21-30-35
screenshot from 2016-05-27 21-30-24
screenshot from 2016-05-27 21-30-10

### CODE REVIEW

**Comment by i386** [comment]
@scherler is the intent that this will look like the mockup?

**Comment by scherler** [comment]
well,  I focused on 100% features, since I am now off to vacation now but you refer to the symbols that are the StatusIndicators and I thought I understood you that we should re-use them.

**Comment by i386** [comment]
@scherler I had misunderstood - I thought you had asked if you could share the components (which is up to Josh and yourself) rather than change the design to fit the available components. We will still need to make this look like the mockup.

**Comment by scherler** [comment]
will fix it later

**Comment by michaelneale** [comment]
@scherler you got any more stuff in the pipe on this, or is it ok for @sophistifunk to start running with it?

**Comment by michaelneale** [comment]
I am mixing metaphors. Pipe -> surf with it?

**Comment by scherler** [comment]
:) The only thing "blank" ATM is the download button (I deactivated it ATM until we know what is the final source of "download") but otherwise it is good to run with. @sophistifunk need to "Link" the graph to the log though that is not done.

**Comment by michaelneale** [comment]
@scherler ok great - no go and do holidays.

File: blueocean-dashboard/src/main/js/components/stories/logDisplay.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { storiesOf } from '@kadira/storybook';
+import Node from '../Node';
+import Nodes from '../Nodes';
+import { getStagesInformation } from '../../util/logDisplayHelper';
+
+import { runNodesRunning, runNodesFail, runNodesSuccess } from '../../../../test/js/runNodes';
```
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/stories/logDisplay.js]: this is very unpleasant.

**Comment by amuniz** [comment]
Wouldn't be _Expand all_ and _Collapse all_ useful? Think of a Pipeline with a hundred steps.

(Asking as I don't see them in the code, but maybe they are there).

File: blueocean-dashboard/src/main/js/components/stories/logDisplay.js (hunk 1 above)
Comment (scherler) [inline blueocean-dashboard/src/main/js/components/stories/logDisplay.js]: can you explain why, please? Keep in mind the above is storybook code.

**Comment by scherler** [comment]
@amuniz https://github.com/jenkinsci/blueocean-plugin/pull/212/files#diff-4f9390c721f57daa3d79657024d0f59fR58 the one you ask for is not in the design @i386 would need to comment on this, but we normally only show one log.

**Comment by i386** [comment]
No expanding or collapsing needed - there's a link to open the full log in a new window

File: blueocean-dashboard/src/main/js/components/stories/logDisplay.js (hunk 1 above)
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/stories/logDisplay.js]: oh - just the ../../ made me cross eyed - as storybook that is ok!

**Comment by michaelneale** [comment]
@amuniz yeah a full log view would make more sense than "expand all". Having said that, over time we can refine as people use things it will become more obvious what is needed.

**Comment by sophistifunk** [comment]
Good to see another instance of `fetch()` bite the dust :D 👍

**Comment by sophistifunk** [comment]
success

**Pull request merged**